### PR TITLE
change to just links for whatsnew index

### PR DESCRIPTION
**Pull request by eteq**
This is a pretty minor thing but I noticed an inconsistency in `docs/whatsnew/index.rst` - it seems to be using two different linking idioms.  This PR corrects the links to effectively point to the "latest" version rather than the "old" versions.

There might have been some reason we left it this way, though?  I honestly can't recall (maybe even I thought it was important? :shrug:) Maybe @bsipocz has some insight on this? (as the last editor of that page)

### CODE REVIEW

**Comment by bsipocz** [comment]
The diversion is when we finally introduced to not have all the what's news (as they started to have broken, and unfixable links in them), and linking with numbers for versions before the change was no possible. So, overall I think would not work, you should certainly double-check all the links in the RTD rendering.

**Comment by eteq** [comment]
Ohhh, I gotcha, thanks @bsipocz !

**Comment by bsipocz** [comment]
One more thing: whatsnew/4.2 content has to be deleted and linked to the released version, similarly to the other versions. And it has to be backported to the release branch

**Comment by bsipocz** [comment]
OK, so having one more look into it, the linking should be the other way around. Only the most recent one should have not a hard-wired link, all the rest should be an orphaned page, and a link to the released version with that version number. 

Apparently, 4.2 hasn't been cleaned up in the 4.3 cycle.

**Review by pllim — CHANGES_REQUESTED** [review]
Yeah, I agree with @bsipocz . This is a step backward. Should be the other way around, I think.

**Comment by eteq** [comment]
Note I elected to not fix this for 4.3rc1 since it seemed more important to get the RC out the door, but leaving in milestoned 4.3 because I am still hoping to get this in as a doc fix for the release.

**Comment by eteq** [comment]
Alright I corrected this.  But it should wait on being merged until after the 4.3 RTD page is generated.

**Comment by pllim** [comment]
I think the commits should be squashed.

**Review by pllim — APPROVED** [review]
LGTM. I will approve to remove merge block and leave it to you, @eteq , to decide when the time is right for merging. Thanks!

**Comment by pllim** [comment]
p.s. I don't think this can be backported. So I am updating the labels and milestone.

**Comment by pllim** [comment]
For the record, this broke RTD. The CI for it did fail. Should not have merged with that failure. FYI.

```
.../docs/whatsnew/4.2.rst: WARNING: document isn't included in any toctree
.../docs/whatsnew/4.3.rst: WARNING: document isn't included in any toctree
```

**Pull request merged**
